lib/Models/WebMapServiceCatalogItem.js

```diff
@@ -319,18 +319,24 @@ WebMapServiceCatalogItem.prototype.updateFromCapabilities = function(capabilitie
 
     if (!defined(thisLayer)) {
         thisLayer = findLayers(capabilities.Capability.Layer, this.layers);
-        if (!defined(thisLayer)) {
-            return;
-        }
 
         if (defined(this.layers)) {
             var layers = this.layers.split(',');
             for (var i = 0; i < thisLayer.length; ++i) {
                 if (!defined(thisLayer[i])) {
-                    console.log('A layer with the name or ID \"' + layers[i] + '\" does not exist on the WMS Server - ignoring it.');
-                    thisLayer.splice(i, 1);
-                    layers.splice(i, 1);
-                    --i;
+                    if (thisLayer.length > 1) {
+                        console.log('A layer with the name or ID \"' + layers[i] + '\" does not exist on the WMS Server - ignoring it.');
+                        thisLayer.splice(i, 1);
+                        layers.splice(i, 1);
+                        --i;
+                    } else {
+                        throw new TerriaError({
+                            title: 'No layer found',
+                            message: 'The WMS dataset "' + this.name + '" has no layers matching "' + this.layers + '". (Perhaps it should be "' + capabilities.Capability.Layer.Layer.Name + '").' + 
```

Comment:
The `capabilities.Capability.Layer.Layer.Name` bit is dangerous.  It's possible (likely?) that one of those dotted parts will evaluate to undefined, and if it does NM will display a really ugly message instead of the nice one you're trying to generate.